pkg/resource/deploy/step_generator.go

```diff
@@ -524,9 +525,14 @@ func (sg *stepGenerator) generateStepsFromDiff(
 			"unrecognized diff state for %s: %d", urn, diff.Changes)
 	}
 
-	// If there were changes, check for a replacement vs. an in-place update.
-	if diff.Changes == plugin.DiffSome {
-		if diff.Replace() {
+	// There are changes if the diff returned DiffSome or if there were initErrors previously that should
+	// force an update to occur even if no inputs changed.
+	needsUpdateOrReplace := diff.Changes == plugin.DiffSome || len(old.InitErrors) > 0
```

Comment:
Changed this to align the initErrors path with the normal property change path - so that initErrors can potentially trigger replacements, but also to make the codepaths a bit more consistent generally.  For example, it looks like there was a potential latent bug where `goal.IgnoreChanges` was not propagated in the initErrors codepath.